### PR TITLE
fix(annotations): use epsilon tolerance for image dimension checks to prevent out-of-bounds errors

### PR DESCRIPTION
**Pull request by bionoone**
### Context

in commit https://github.com/cornerstonejs/cornerstone3D/commit/20efd77303d874f1c0a59f3cd859bdbce47d4a67#diff-4bdf7112153ce30f1829f12a4e4899bcf36b964d578c23c5c0e82235f8e166f3

there is a change in convertion from transformWorldToIndex that perform a integer index conversion in favor of imageData.worldToIndex that perform continuous index conversion.

This has an impact on annotation drawn on stack viewport, where z extend is from 0 to 1.

due to conversion, the z coordinate could become something -1e-5 leading to some side effect when preventHandleOutsideImage  option is on.

### Changes & Results

added a tollerance of EPSILON in new method indexAlmostWithinDimensions 

### Testing

add karma test exhibiting the defect

#### PR

- [x] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [x] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [ ] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [ ] "OS: 
- [ ] "Node version: 
- [ ] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

## Summary by CodeRabbit

* **New Features**
  * Added a more tolerant “almost within bounds” utility for dimension checks.
  * Expanded tool volume detection to use this tolerance, improving edge-case handling near boundaries.

* **Bug Fixes**
  * Drawing annotations near image edges now behaves more consistently, reducing unwanted deletions when points are only slightly outside bounds.

* **Tests**
  * Added coverage for annotation behavior both inside and outside the image area.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1 +1 @@
-yarn lint-staged
+bun lint-staged
```
Comment (sedghi) [inline .husky/pre-commit]: please revert

Comment (bionoone) [reply .husky/pre-commit]: done

**Review by sedghi — APPROVED** [review]
Thanks
